modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BTypes.java

```diff
@@ -145,9 +145,15 @@ public static BType resolveType(SimpleTypeName typeName, SymbolScope symbolScope
         // If bType is not null, then element type of this arrays type is available.
         // We should define the arrays type here.
         if (bType != null) {
+            SimpleTypeName childSimpleType = new SimpleTypeName(typeName.getName(), typeName.getPackagePath(), true);
+            childSimpleType.setPkgPath(typeName.getPackagePath());
+            childSimpleType.setArrayType(typeName.getDimensions() - 1);
```

Comment:
Check whether typeName.getDimensions() is equal to 1, if so don't create a type name